qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1420,6 +1422,37 @@ def collect_runs(self, namelist):
                     group_list.append(tuple(group))
         return set(group_list)
 
+    def collect_1q_runs(self):
+        """Return a set of non-conditional runs of 1q "op" nodes."""
+        group_list = []
+        # Iterate through the nodes of self in topological order
+        # and form tuples containing sequences of gates
+        # on the same qubit(s).
+        topo_ops = list(self.topological_op_nodes())
```

Comment:
The `list` looks like it isn't needed.

Reply:
nope it's probably not when coupled with your below suggestion (topological_op_nodes() returns an iter iirc). But, I just copied the implementation from `collect_runs()`. Since this is all not present on master anymore I was just trying to reduce the risk of a regression by using basically the same code with a different condition.

Reply:
Agree with trying to reduce the risk of regression. This one might be worth fixing though, both because it's small and because the iter -> list (combined with the list -> iter in `DAGCircuit.topological_op_nodes`) will end up generating an unneeded (shallow) copy of the op nodes list.

Reply:
Ok, done in: https://github.com/Qiskit/qiskit-terra/pull/5655/commits/b202cb4db48b26177602d91a40cd3a3ffd4c52b6